outport/process/executionOrder/transactionsExecutionOrder.go

```diff
@@ -0,0 +1,258 @@
+
+	// scheduled to me, already sorted
+	scheduledTransactionsToMe := extractNormalTransactionAndScrsToMe(pool, blockBody, header, false)
+
+	// need to be sorted
+	transactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, true)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(transactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	rewardsTxs := getRewardsTxsFromMe(pool, blockBody, header)
+
+	// scheduled from me, need to be sorted
+	scheduledTransactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, false)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(scheduledTransactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	allTransaction := append(transactionsToMe, scheduledTransactionsToMe...)
+	allTransaction = append(allTransaction, transactionsFromMe...)
+	allTransaction = append(allTransaction, rewardsTxs...)
+	allTransaction = append(allTransaction, scheduledTransactionsFromMe...)
+
+	for idx, tx := range allTransaction {
+		tx.SetExecutionOrder(idx)
+	}
+
+	setOrderSmartContractResults(pool)
+
+	prinPool(pool)
+	return nil
+}
+
+func setOrderSmartContractResults(pool *outport.Pool) {
+	for _, scrHandler := range pool.Scrs {
+		scr, ok := scrHandler.GetTxHandler().(*smartContractResult.SmartContractResult)
+		if !ok {
+			continue
+		}
+
+		tx, found := pool.Txs[string(scr.OriginalTxHash)]
+		if !found {
+			continue
+		}
+
+		scrHandler.SetExecutionOrder(tx.GetExecutionOrder())
+	}
+}
+
+func extractNormalTransactionsAndInvalidFromMe(pool *outport.Pool, blockBody *block.Body, header data.HeaderHandler, ignoreScheduled bool) []data.TransactionHandlerWithGasUsedAndFee {
+	normalTxsHashes := make([][]byte, 0)
+	invalidTxsHashes := make([][]byte, 0)
+	for mbIndex, mb := range blockBody.MiniBlocks {
+		if isMBScheduled(header, mbIndex) == ignoreScheduled || shouldIgnoreProcessedMBScheduled(header, mbIndex) {
+			continue
+		}
+
+		isFromMe := mb.SenderShardID == header.GetShardID()
+		if !isFromMe {
+			continue
+		}
+
+		if mb.Type == block.TxBlock {
+			normalTxsHashes = append(normalTxsHashes, mb.TxHashes...)
+			continue
+		}
+		if mb.Type == block.InvalidBlock {
+			invalidTxsHashes = append(invalidTxsHashes, mb.TxHashes...)
+			continue
+		}
+	}
+
+	normalTxs := extractTxsFromMap(normalTxsHashes, pool.Txs)
+	invalidTxs := extractTxsFromMap(invalidTxsHashes, pool.Invalid)
+
+	return append(normalTxs, invalidTxs...)
+}
+
+func extractNormalTransactionAndScrsToMe(pool *outport.Pool, blockBody *block.Body, header data.HeaderHandler, ignoreScheduled bool) []data.TransactionHandlerWithGasUsedAndFee {
+	grouped := make([]data.TransactionHandlerWithGasUsedAndFee, 0)
+	for mbIndex, mb := range blockBody.MiniBlocks {
+		if isMBScheduled(header, mbIndex) == ignoreScheduled || shouldIgnoreProcessedMBScheduled(header, mbIndex) {
```

Comment:
same here, please remove the ignoreScheduled parameter

Reply:
done